static/src/javascripts/projects/common/modules/identity/upsell/consent-card/FollowCardList.js

```diff
@@ -1,41 +1,69 @@
 // @flow
-import React, { Component } from 'preact-compat';
+import React from 'preact-compat';
 import config from 'lib/config';
 import { FollowCard } from 'common/modules/identity/upsell/consent-card/FollowCard';
+import {
+    setConsentsInApi,
+    getNewsLetterConsents,
+    getUserConsents,
+} from 'common/modules/identity/upsell/store/consents';
 import type { ConsentWithState } from '../store/types';
-import { setConsentsInApi } from '../store/consents';
 import { ErrorBar, genericErrorStr } from '../error-bar/ErrorBar';
 import { ExpanderButton } from '../button/ExpanderButton';
 
-type FollowCardListProps = {
-    consents: Promise<ConsentWithState>[],
+const getConsents = (): Promise<ConsentWithState[]> => {
+    const userConsents = getUserConsents([
+        'supporter',
+        'holidays',
+        'events',
+        'offers',
+        'jobs',
+    ]);
+
+    const newsLetterConsents = getNewsLetterConsents([
+        'today-uk',
+        'the-long-read',
+        'bookmarks',
+        'brexit-briefing',
+        'green-light',
+        'lab-notes',
+    ]);
+
+    return Promise.all([userConsents, newsLetterConsents]).then(consents =>
+        consents[0].concat(consents[1])
+    );
+};
+
+type Props = {
     cutoff: number,
 };
 
-class FollowCardList extends Component<
-    FollowCardListProps,
-    {
-        consents: ConsentWithState[],
-        isExpanded: boolean,
-    }
-> {
-    constructor(props: FollowCardListProps) {
+type State = {
+    consents: ConsentWithState[],
```

Comment:
For now make `FollowCardList` responsible for obtaining consents. I didn't see a reason for passing through consents as a promise from the parent component. Although as alluded to in a comment, I'd like to separate this component into stateful and stateless constituents in another PR.

Reply:
the idea was indeed to have this component be presentational and not know about the cards that it would display, which would in theory make it a bit more resilient to AB tests and potentially personalising the cards temselves. There's nothing preventing that from happening inside but as per the comment the ideal state would be to have two separate components 👍